<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="heinin" />

<meta name="date" content="2024-02-23" />

<title>Annotating cell types in Kluc tumors</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PD1_mm</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/heinin/PD1_mm">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Annotating cell types in Kluc tumors</h1>
<h4 class="author">heinin</h4>
<h4 class="date">2024-02-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>PD1_mm/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240223code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240223)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240223code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240223)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/home/hnatri/PD1_mm/
</td>
<td style="text-align:left;">
.
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PD1_mm/code/utilities.R
</td>
<td style="text-align:left;">
code/utilities.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PD1_mm/code/PD1_mm_themes.R
</td>
<td style="text-align:left;">
code/PD1_mm_themes.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PD1_mm/code/CART_plot_functions.R
</td>
<td style="text-align:left;">
code/CART_plot_functions.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PD1_mm/cluster_markers.tsv
</td>
<td style="text-align:left;">
cluster_markers.tsv
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PD1_mm/analysis/annotate.Rmd
</td>
<td style="text-align:left;">
analysis/annotate.Rmd
</td>
</tr>
<tr>
<td style="text-align:left;">
/home/hnatri/PD1_mm/docs/Layer1_scImmuCC_label.csv
</td>
<td style="text-align:left;">
docs/Layer1_scImmuCC_label.csv
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomheininPD1mmtree3e207b9b95b4dc3943413f3ed48ff5834b1b9f0ctargetblank3e207b9a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/heinin/PD1_mm/tree/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c" target="_blank">3e207b9</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomheininPD1mmtree3e207b9b95b4dc3943413f3ed48ff5834b1b9f0ctargetblank3e207b9a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/heinin/PD1_mm/tree/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c" target="_blank">3e207b9</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/annotate.Rmd
    Modified:   analysis/comparative_analysis.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/annotate.Rmd</code>) and HTML
(<code>docs/annotate.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c/analysis/annotate.Rmd" target="_blank">3e207b9</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
<td>
Added scImmuCC annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/heinin/PD1_mm/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c/docs/annotate.html" target="_blank">3e207b9</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
<td>
Added scImmuCC annotations
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/407ac358c929413de43e478575181ab95e6c2e15/analysis/annotate.Rmd" target="_blank">407ac35</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
<td>
Updating the first script
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/heinin/PD1_mm/407ac358c929413de43e478575181ab95e6c2e15/docs/annotate.html" target="_blank">407ac35</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
<td>
Updating the first script
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/196db6ad343a2ed3738607de33734bb5d1f6ca20/analysis/annotate.Rmd" target="_blank">196db6a</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
<td>
Starting the comparative analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/heinin/PD1_mm/196db6ad343a2ed3738607de33734bb5d1f6ca20/docs/annotate.html" target="_blank">196db6a</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
<td>
Starting the comparative analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/627f4bfdf2899a0847ba3638b71b9521980204b5/analysis/annotate.Rmd" target="_blank">627f4bf</a>
</td>
<td>
heinin
</td>
<td>
2024-02-23
</td>
<td>
Updated scripts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/heinin/PD1_mm/627f4bfdf2899a0847ba3638b71b9521980204b5/docs/annotate.html" target="_blank">627f4bf</a>
</td>
<td>
heinin
</td>
<td>
2024-02-23
</td>
<td>
Updated scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/9e080d718b833fcdaac206af2a172849dc57bde6/analysis/annotate.Rmd" target="_blank">9e080d7</a>
</td>
<td>
heinin
</td>
<td>
2024-02-23
</td>
<td>
Added the initial annotation script
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/heinin/PD1_mm/9e080d718b833fcdaac206af2a172849dc57bde6/docs/annotate.html" target="_blank">9e080d7</a>
</td>
<td>
heinin
</td>
<td>
2024-02-23
</td>
<td>
Added the initial annotation script
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Initial analysis on the scRNAseq data from Kluc tumors treated with
PD1 and/or CAR T.</p>
<div id="packages-and-environment-variables" class="section level3">
<h3>Packages and environment variables</h3>
<pre class="r"><code>suppressPackageStartupMessages({
  #library(cli)
  library(Seurat)
  library(SeuratObject)
  library(SeuratDisk)
  library(tidyverse)
  library(tibble)
  library(ggplot2)
  library(ggpubr)
  library(ggrepel)
  library(workflowr)
  library(googlesheets4)
  library(scImmuCC)})

setwd(&quot;/home/hnatri/PD1_mm/&quot;)
set.seed(9999)
options(ggrepel.max.overlaps = Inf)

# Colors, themes, cell type markers, and plot functions
source(&quot;/home/hnatri/PD1_mm/code/utilities.R&quot;)
source(&quot;/home/hnatri/PD1_mm/code/PD1_mm_themes.R&quot;)
source(&quot;/home/hnatri/PD1_mm/code/CART_plot_functions.R&quot;)</code></pre>
</div>
<div id="importing-data" class="section level3">
<h3>Importing data</h3>
<pre class="r"><code>seurat_data &lt;- readRDS(&quot;/tgen_labs/banovich/BCTCSF/PD1_mm_Seurat/PD1_mm_Seurat_merged.Rds&quot;)</code></pre>
</div>
<div id="adding-human-gene-names-to-a-separate-assay"
class="section level3">
<h3>Adding human gene names to a separate assay</h3>
<pre class="r"><code># Converting mouse gene names to human
mouse_human_genes &lt;- read.csv(&quot;http://www.informatics.jax.org/downloads/reports/HOM_MouseHumanSequence.rpt&quot;, sep=&quot;\t&quot;)

convert_mouse_to_human &lt;- function(gene_list){
  gene_names &lt;- as.data.frame(matrix(nrow = length(gene_list),
                                     ncol = 2))
  colnames(gene_names) &lt;- c(&quot;mouse&quot;, &quot;human&quot;)
  rownames(gene_names) &lt;- gene_list
  gene_names$mouse &lt;- gene_list
  
  for(gene in gene_list){
    class_key = (mouse_human_genes %&gt;% filter(Symbol == gene &amp; Common.Organism.Name==&quot;mouse, laboratory&quot;))[[&#39;DB.Class.Key&#39;]]
    if(!identical(class_key, integer(0)) ){
      human_genes = (mouse_human_genes %&gt;% filter(DB.Class.Key == class_key &amp; Common.Organism.Name==&quot;human&quot;))[,&quot;Symbol&quot;]
      
      if(length(human_genes)==0){
        gene_names[gene, &quot;human&quot;] &lt;- NA
      } else if (length(human_genes)&gt;1){
        #  human_genes &lt;- paste0(human_genes, collapse = &quot;, &quot;)
        bind_df &lt;- data.frame(&quot;mouse&quot; = rep(gene, times = length(human_genes)),
                              &quot;human&quot; = human_genes)
        gene_names &lt;- rbind(gene_names, bind_df)
      } else {
        gene_names[gene, &quot;human&quot;] &lt;- human_genes
      }
    }
  }
  return(gene_names)
}

gene_names &lt;- convert_mouse_to_human(rownames(seurat_data@assays$RNA))</code></pre>
<pre><code>Warning: There was 1 warning in `filter()`.
ℹ In argument: `DB.Class.Key == class_key &amp; Common.Organism.Name == &quot;human&quot;`.
Caused by warning in `DB.Class.Key == class_key`:
! longer object length is not a multiple of shorter object length</code></pre>
<pre class="r"><code>length(rownames(seurat_data@assays$RNA))</code></pre>
<pre><code>[1] 32285</code></pre>
<pre class="r"><code>dim(gene_names)</code></pre>
<pre><code>[1] 36196     2</code></pre>
<pre class="r"><code># Keeping mouse genes with a single human ortholog
gene_names &lt;- gene_names %&gt;%
  group_by(mouse) %&gt;%
  filter(!is.na(human),
         n() == 1) %&gt;%
  ungroup()

assay_data &lt;- LayerData(seurat_data, assay = &quot;RNA&quot;, layer = &quot;counts&quot;)
assay_data &lt;- assay_data[which(rownames(assay_data) %in% gene_names$mouse),]
new_names &lt;- rownames(assay_data)
new_names &lt;- mapvalues(x = new_names,
                       from = gene_names$mouse,
                       to = gene_names$human)
rownames(assay_data) &lt;- new_names

seurat_data[[&quot;RNA_human&quot;]] &lt;- CreateAssayObject(assay_data,
                                               min.cells = 0,
                                               min.features = 0)</code></pre>
<pre><code>Warning: Non-unique features (rownames) present in the input matrix, making
unique</code></pre>
<pre class="r"><code>#saveRDS(seurat_data, &quot;/tgen_labs/banovich/BCTCSF/PD1_mm_Seurat/PD1_mm_Seurat_merged.Rds&quot;)</code></pre>
</div>
<div id="visualizing-clusters" class="section level3">
<h3>Visualizing clusters</h3>
<pre class="r"><code>DimPlot(seurat_data,
        group.by = &quot;snn_res.0.5&quot;,
        reduction = &quot;umap&quot;,
        raster = T,
        cols = cluster_col,
        label = T) &amp;
  coord_fixed(ratio = 1) &amp;
  theme_classic() &amp;
  NoLegend() &amp;
  manuscript_theme</code></pre>
<p><img src="figure/annotate.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c/docs/figure/annotate.Rmd/unnamed-chunk-4-1.png" target="_blank">3e207b9</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-type-marker-expression" class="section level3">
<h3>Cell type marker expression</h3>
<pre class="r"><code>DefaultAssay(seurat_data) &lt;- &quot;RNA&quot;

plot_features &lt;- c()

# Top markers for each cluster
markers &lt;- presto::wilcoxauc(seurat_data,
                             group_by = &quot;snn_res.0.5&quot;,
                             assay = &quot;data&quot;,
                             seurat_assay = &quot;RNA&quot;)

top_markers &lt;- markers %&gt;%  group_by(group) %&gt;% slice_max(order_by = auc, n = 2)

FeaturePlot(seurat_data,
            features = top_markers$feature,
            ncol = 5,
            reduction = &quot;umap&quot;,
            raster = T,
            cols = c(&quot;gray89&quot;, &quot;tomato3&quot;)) &amp;
  coord_fixed(ratio = 1) &amp;
  theme_classic() &amp;
  NoLegend() &amp;
  manuscript_theme</code></pre>
<p><img src="figure/annotate.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c/docs/figure/annotate.Rmd/unnamed-chunk-5-1.png" target="_blank">3e207b9</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>top_markers &lt;- markers %&gt;%  group_by(group) %&gt;% slice_max(order_by = auc, n = 5)

# seurat_object, plot_features, group_var, group_colors, column_title, km=5, row.order = NULL
dotplot_heatmap &lt;- create_dotplot_heatmap(seurat_object = seurat_data,
                                          plot_features = unique(top_markers$feature),
                                          group_var = &quot;snn_res.0.5&quot;,
                                          group_colors = cluster_col,
                                          column_title = &quot;&quot;,
                                          km = 5, row.order = NULL)</code></pre>
<p><img src="figure/annotate.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c/docs/figure/annotate.Rmd/unnamed-chunk-6-1.png" target="_blank">3e207b9</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="saving-top-markers-to-a-file" class="section level3">
<h3>Saving top markers to a file</h3>
<pre class="r"><code>top_markers &lt;- markers %&gt;%  group_by(group) %&gt;% slice_max(order_by = auc, n = 20)

write.table(top_markers, &quot;/home/hnatri/PD1_mm/cluster_markers.tsv&quot;,
            quote = F, row.names = F, sep = &quot;\t&quot;)

# To build on command line, run Rscript -e &quot;rmarkdown::render(&#39;/home/hnatri/PD1_mm/analysis/annotate.Rmd&#39;)&quot;
# Then &quot;mv /home/hnatri/PD1_mm/analysis/annotate.Rmd /home/hnatri/PD1_mm/docs/&quot;</code></pre>
</div>
<div id="canonical-mouse-immune-markers" class="section level3">
<h3>Canonical mouse immune markers</h3>
<pre class="r"><code># Mouse immune markers
gs4_deauth()
canonical_markers  &lt;- gs4_get(&quot;https://docs.google.com/spreadsheets/d/1ApwXjEVtpPB87al6q3ab8TKvZYJTh3iNH1cuO-A_OoU/edit?usp=sharing&quot;)
sheet_names(canonical_markers)</code></pre>
<pre><code> [1] &quot;Sample summary&quot;                                   
 [2] &quot;GSEA GBM&quot;                                         
 [3] &quot;Mm immune markers&quot;                                
 [4] &quot;Cluster annotations, JAK mouse&quot;                   
 [5] &quot;Sherri-Cluster annotations, JAK mouse&quot;            
 [6] &quot;Cluster markers, JAK mouse&quot;                       
 [7] &quot;Sherri-Cluster markers, JAK mouse&quot;                
 [8] &quot;DL_cluster_res&quot;                                   
 [9] &quot;Cluster markers, final data, UPN109pre, no UPN208&quot;
[10] &quot;Cluster annotations, GBM+mouse&quot;                   
[11] &quot;Cluster markers, GBM+JAK1KO&quot;                      
[12] &quot;Sherri-Cluster markers, GBM+JAK1KO&quot;               
[13] &quot;Cluster markers, final data, immune+fibroblast&quot;   
[14] &quot;Cluster annotations, immune+fibroblast&quot;           
[15] &quot;Cluster markers, immune+fibroblast, top 100&quot;      
[16] &quot;Sherri - Cluster annotations, immune+fibroblast&quot;  
[17] &quot;Sherri-Only GBM Charactarization 50 genes&quot;        
[18] &quot;Sherri only GBM immune+fibroblast, top 100&quot;       
[19] &quot;Cluster annotations&quot;                              
[20] &quot;Heatmap genes&quot;                                    
[21] &quot;Tumor and CSF sample summary&quot;                     </code></pre>
<pre class="r"><code>mm_immune_markers &lt;- read_sheet(canonical_markers, sheet = &quot;Mm immune markers&quot;)</code></pre>
<pre><code>✔ Reading from &quot;13384 tumor scRNAseq tables&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Mm immune markers&#39;&#39;.</code></pre>
<pre class="r"><code>dotplot_heatmap &lt;- create_dotplot_heatmap(seurat_object = seurat_data,
                                          plot_features = mm_immune_markers$gene_name,
                                          group_var = &quot;snn_res.0.5&quot;,
                                          group_colors = cluster_col,
                                          column_title = &quot;&quot;,
                                          km = 5, row.order = NULL)</code></pre>
<pre><code>Warning: The following requested variables were not found: Cd51, Nirp3, 117r</code></pre>
<p><img src="figure/annotate.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c/docs/figure/annotate.Rmd/unnamed-chunk-8-1.png" target="_blank">3e207b9</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="using-sc-immucc-to-annotate-immune-cells"
class="section level3">
<h3>Using sc-ImmuCC to annotate immune cells</h3>
<pre class="r"><code>count_data &lt;- LayerData(seurat_data, assay = &quot;RNA_human&quot;, layer = &quot;counts&quot;)

#scImmuCC_Layered(count = count_data, Non_Immune = FALSE)

# Importing results
scicc_labels &lt;- read.csv(&quot;/home/hnatri/PD1_mm/docs/Layer1_scImmuCC_label.csv&quot;,
                         row.names = &quot;X&quot;)

length(colnames(seurat_data))
length(intersect(scicc_labels$barcodes, colnames(seurat_data)))

seurat_data$scImmuCC_celltype &lt;- mapvalues(x = colnames(seurat_data),
                                           from = scicc_labels$barcodes,
                                           to = scicc_labels$cell_type)

# Plotting
DimPlot(seurat_data,
        group.by = &quot;scImmuCC_celltype&quot;,
        reduction = &quot;umap&quot;,
        raster = T,
        #cols = scImmuCC_celltype_col,
        label = T) &amp;
  coord_fixed(ratio = 1) &amp;
  theme_classic() &amp;
  NoLegend() &amp;
  manuscript_theme</code></pre>
<p><img src="figure/annotate.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c/docs/figure/annotate.Rmd/unnamed-chunk-9-1.png" target="_blank">3e207b9</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#saveRDS(seurat_data, &quot;/tgen_labs/banovich/BCTCSF/PD1_mm_Seurat/PD1_mm_Seurat_merged.Rds&quot;)</code></pre>
<pre class="r"><code>DimPlot(seurat_data,
        split.by = &quot;scImmuCC_celltype&quot;,
        group.by = &quot;snn_res.0.5&quot;,
        ncol = 3,
        reduction = &quot;umap&quot;,
        raster = T,
        cols = cluster_col) &amp;
  coord_fixed(ratio = 1) &amp;
  theme_classic() &amp;
  NoLegend()</code></pre>
<p><img src="figure/annotate.Rmd/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/heinin/PD1_mm/blob/3e207b9b95b4dc3943413f3ed48ff5834b1b9f0c/docs/figure/annotate.Rmd/unnamed-chunk-10-1.png" target="_blank">3e207b9</a>
</td>
<td>
heinin
</td>
<td>
2024-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>table(seurat_data$scImmuCC_celltype,
      seurat_data$snn_res.0.5)</code></pre>
<pre><code>            
                0    1    2    3    4    5    6    7    8    9   10   11   12
  Bcell         0    2    0    1    0    2    0    0    2    1    4 1509    2
  DC            5   41   33   17    3    7   68    0   85    7    4    6  715
  ILC           2    0    0    0    0    0    0    0    0    0    0    1    2
  Macrophage  370 3831 2258 1048   67 1306 1056  136 2077 2573 1457  122  608
  Mast        143    2    2    0    4    0    1    7   23    3   78    3    2
  Monocyte     27  706 2013 3122    7 1231 2057   13  716  120  529   50  314
  Neutrophil   82    0    1    0    3    0    0   16  123    0  449    8   21
  NK          193    0    0    0  743   99    0 2891   21    1   26   11    3
  Tcell      3966   11    0    0 2553  543    2   67   17    4   29   42    5
            
               13   14   15   16   17   18   19
  Bcell         0    0    4    2  107    0    0
  DC           15   95  252   32   34    2    0
  ILC           0    0    3    0   36    0    0
  Macrophage 1097  603  666  929  324  343    0
  Mast          2    1    1   10    1    4    0
  Monocyte    524  373  156  128   64   22    2
  Neutrophil   10    1   87   25    2    7    0
  NK            5   58   14   16    8   13   14
  Tcell         5  263   78   12   45    6  218</code></pre>
</div>
<div id="adding-annotations-from-the-google-sheet"
class="section level3">
<h3>Adding annotations from the Google Sheet</h3>
<pre class="r"><code>gs4_deauth()
markers_annotations  &lt;- gs4_get(&quot;https://docs.google.com/spreadsheets/d/1iWYBouwQlQboI-rwiujC0QKJ6lq9XeTffbKm2Nz8es0/edit?usp=sharing&quot;)
sheet_names(markers_annotations)</code></pre>
<pre><code>[1] &quot;Cluster top markers&quot; &quot;Cluster annotations&quot; &quot;scImmuCC&quot;           
[4] &quot;Mm immune markers&quot;  </code></pre>
<pre class="r"><code>annotations &lt;- read_sheet(markers_annotations, sheet = &quot;Cluster annotations&quot;)</code></pre>
<pre><code>✔ Reading from &quot;PD1 mm scRNAseq tables&quot;.</code></pre>
<pre><code>✔ Range &#39;&#39;Cluster annotations&#39;&#39;.</code></pre>
<pre class="r"><code>seurat_data$celltype &lt;- mapvalues(seurat_data$snn_res.0.5,
                                  from = annotations$snn_res.0.5,
                                  to = annotations$annotation)

DimPlot(seurat_data,
        group.by = &quot;celltype&quot;,
        reduction = &quot;umap&quot;,
        raster = T,
        label = T) &amp;
  coord_fixed(ratio = 1) &amp;
  theme_classic() &amp;
  NoLegend()</code></pre>
<p><img src="figure/annotate.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>saveRDS(seurat_data, &quot;/tgen_labs/banovich/BCTCSF/PD1_mm_Seurat/PD1_mm_Seurat_merged.Rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ComplexHeatmap_2.18.0 viridis_0.6.3         viridisLite_0.4.2    
 [4] circlize_0.4.15       plyr_1.8.8            RColorBrewer_1.1-3   
 [7] scImmuCC_1.0.0        GSVA_1.50.0           googlesheets4_1.1.0  
[10] workflowr_1.7.1       ggrepel_0.9.3         ggpubr_0.6.0         
[13] lubridate_1.9.2       forcats_1.0.0         stringr_1.5.0        
[16] dplyr_1.1.2           purrr_1.0.1           readr_2.1.4          
[19] tidyr_1.3.0           tibble_3.2.1          ggplot2_3.4.2        
[22] tidyverse_2.0.0       SeuratDisk_0.0.0.9021 Seurat_5.0.1         
[25] SeuratObject_5.0.1    sp_1.6-1             

loaded via a namespace (and not attached):
  [1] fs_1.6.2                    matrixStats_1.0.0          
  [3] spatstat.sparse_3.0-1       bitops_1.0-7               
  [5] doParallel_1.0.17           httr_1.4.6                 
  [7] tools_4.3.0                 sctransform_0.4.1          
  [9] backports_1.4.1             utf8_1.2.3                 
 [11] R6_2.5.1                    HDF5Array_1.30.0           
 [13] lazyeval_0.2.2              uwot_0.1.14                
 [15] GetoptLong_1.0.5            rhdf5filters_1.14.1        
 [17] withr_2.5.0                 gridExtra_2.3              
 [19] progressr_0.13.0            cli_3.6.1                  
 [21] Biobase_2.62.0              Cairo_1.6-0                
 [23] spatstat.explore_3.2-1      fastDummies_1.7.3          
 [25] labeling_0.4.2              sass_0.4.6                 
 [27] spatstat.data_3.0-1         ggridges_0.5.4             
 [29] pbapply_1.7-0               parallelly_1.36.0          
 [31] rstudioapi_0.14             RSQLite_2.3.1              
 [33] shape_1.4.6                 generics_0.1.3             
 [35] ica_1.0-3                   spatstat.random_3.1-5      
 [37] car_3.1-2                   Matrix_1.6-5               
 [39] fansi_1.0.4                 S4Vectors_0.40.2           
 [41] abind_1.4-5                 lifecycle_1.0.3            
 [43] whisker_0.4.1               yaml_2.3.7                 
 [45] carData_3.0-5               SummarizedExperiment_1.32.0
 [47] rhdf5_2.46.1                SparseArray_1.2.3          
 [49] Rtsne_0.16                  blob_1.2.4                 
 [51] promises_1.2.0.1            crayon_1.5.2               
 [53] miniUI_0.1.1.1              lattice_0.21-8             
 [55] beachmat_2.18.0             cowplot_1.1.1              
 [57] annotate_1.80.0             KEGGREST_1.42.0            
 [59] magick_2.7.4                pillar_1.9.0               
 [61] knitr_1.43                  GenomicRanges_1.54.1       
 [63] rjson_0.2.21                future.apply_1.11.0        
 [65] codetools_0.2-19            leiden_0.4.3               
 [67] glue_1.6.2                  getPass_0.2-4              
 [69] data.table_1.14.8           vctrs_0.6.2                
 [71] png_0.1-8                   spam_2.9-1                 
 [73] cellranger_1.1.0            gtable_0.3.3               
 [75] cachem_1.0.8                xfun_0.39                  
 [77] S4Arrays_1.2.0              mime_0.12                  
 [79] survival_3.5-5              gargle_1.4.0               
 [81] SingleCellExperiment_1.24.0 iterators_1.0.14           
 [83] ellipsis_0.3.2              fitdistrplus_1.1-11        
 [85] ROCR_1.0-11                 nlme_3.1-162               
 [87] bit64_4.0.5                 RcppAnnoy_0.0.20           
 [89] GenomeInfoDb_1.38.5         rprojroot_2.0.3            
 [91] bslib_0.4.2                 irlba_2.3.5.1              
 [93] KernSmooth_2.23-21          colorspace_2.1-0           
 [95] BiocGenerics_0.48.1         DBI_1.1.3                  
 [97] tidyselect_1.2.0            processx_3.8.1             
 [99] curl_5.0.0                  bit_4.0.5                  
[101] compiler_4.3.0              git2r_0.32.0               
[103] graph_1.80.0                hdf5r_1.3.8                
[105] DelayedArray_0.28.0         plotly_4.10.2              
[107] scales_1.2.1                lmtest_0.9-40              
[109] callr_3.7.3                 digest_0.6.31              
[111] goftest_1.2-3               presto_1.0.0               
[113] spatstat.utils_3.0-3        rmarkdown_2.22             
[115] XVector_0.42.0              htmltools_0.5.5            
[117] pkgconfig_2.0.3             sparseMatrixStats_1.14.0   
[119] MatrixGenerics_1.14.0       highr_0.10                 
[121] fastmap_1.1.1               GlobalOptions_0.1.2        
[123] rlang_1.1.1                 htmlwidgets_1.6.2          
[125] shiny_1.7.4                 DelayedMatrixStats_1.24.0  
[127] farver_2.1.1                jquerylib_0.1.4            
[129] zoo_1.8-12                  jsonlite_1.8.5             
[131] BiocParallel_1.36.0         BiocSingular_1.18.0        
[133] RCurl_1.98-1.12             magrittr_2.0.3             
[135] GenomeInfoDbData_1.2.11     dotCall64_1.0-2            
[137] patchwork_1.1.2             Rhdf5lib_1.24.1            
[139] munsell_0.5.0               Rcpp_1.0.10                
[141] reticulate_1.29             stringi_1.7.12             
[143] zlibbioc_1.48.0             MASS_7.3-60                
[145] parallel_4.3.0              listenv_0.9.0              
[147] deldir_1.0-9                Biostrings_2.70.1          
[149] splines_4.3.0               tensor_1.5                 
[151] hms_1.1.3                   ps_1.7.5                   
[153] igraph_1.4.3                spatstat.geom_3.2-1        
[155] ggsignif_0.6.4              RcppHNSW_0.5.0             
[157] reshape2_1.4.4              stats4_4.3.0               
[159] ScaledMatrix_1.10.0         XML_3.99-0.14              
[161] evaluate_0.21               foreach_1.5.2              
[163] tzdb_0.4.0                  httpuv_1.6.11              
[165] RANN_2.6.1                  polyclip_1.10-4            
[167] clue_0.3-64                 future_1.32.0              
[169] scattermore_1.2             rsvd_1.0.5                 
[171] broom_1.0.4                 xtable_1.8-4               
[173] RSpectra_0.16-1             rstatix_0.7.2              
[175] later_1.3.1                 googledrive_2.1.0          
[177] memoise_2.0.1               AnnotationDbi_1.64.1       
[179] IRanges_2.36.0              cluster_2.1.4              
[181] timechange_0.2.0            globals_0.16.2             
[183] GSEABase_1.64.0            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
