---
title: "Annotating cell types in Kluc tumors"
author: "heinin"
date: "2024-02-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

Initial analysis on the scRNAseq data from Kluc tumors treated with PD1 and/or
CAR T.

### Packages and environment variables

```{r, warning=F, message=F}

suppressPackageStartupMessages({
  #library(cli)
  library(Seurat)
  library(SeuratObject)
  library(SeuratDisk)
  library(tidyverse)
  library(tibble)
  library(ggplot2)
  library(ggpubr)
  library(ggrepel)
  library(workflowr)
  library(googlesheets4)
  library(scImmuCC)})

setwd("/home/hnatri/PD1_mm/")
set.seed(9999)
options(ggrepel.max.overlaps = Inf)

# Colors, themes, cell type markers, and plot functions
source("/home/hnatri/PD1_mm/code/utilities.R")
source("/home/hnatri/PD1_mm/code/PD1_mm_themes.R")
source("/home/hnatri/PD1_mm/code/CART_plot_functions.R")

```

### Importing data

```{r}

seurat_data <- readRDS("/tgen_labs/banovich/BCTCSF/PD1_mm_Seurat/PD1_mm_Seurat_merged.Rds")

```

### Adding human gene names to a separate assay

```{r}

# Converting mouse gene names to human
mouse_human_genes <- read.csv("http://www.informatics.jax.org/downloads/reports/HOM_MouseHumanSequence.rpt", sep="\t")

convert_mouse_to_human <- function(gene_list){
  gene_names <- as.data.frame(matrix(nrow = length(gene_list),
                                     ncol = 2))
  colnames(gene_names) <- c("mouse", "human")
  rownames(gene_names) <- gene_list
  gene_names$mouse <- gene_list
  
  for(gene in gene_list){
    class_key = (mouse_human_genes %>% filter(Symbol == gene & Common.Organism.Name=="mouse, laboratory"))[['DB.Class.Key']]
    if(!identical(class_key, integer(0)) ){
      human_genes = (mouse_human_genes %>% filter(DB.Class.Key == class_key & Common.Organism.Name=="human"))[,"Symbol"]
      
      if(length(human_genes)==0){
        gene_names[gene, "human"] <- NA
      } else if (length(human_genes)>1){
        #  human_genes <- paste0(human_genes, collapse = ", ")
        bind_df <- data.frame("mouse" = rep(gene, times = length(human_genes)),
                              "human" = human_genes)
        gene_names <- rbind(gene_names, bind_df)
      } else {
        gene_names[gene, "human"] <- human_genes
      }
    }
  }
  return(gene_names)
}

gene_names <- convert_mouse_to_human(rownames(seurat_data@assays$RNA))

length(rownames(seurat_data@assays$RNA))
dim(gene_names)

# Keeping mouse genes with a single human ortholog
gene_names <- gene_names %>%
  group_by(mouse) %>%
  filter(!is.na(human),
         n() == 1) %>%
  ungroup()

assay_data <- LayerData(seurat_data, assay = "RNA", layer = "counts")
assay_data <- assay_data[which(rownames(assay_data) %in% gene_names$mouse),]
new_names <- rownames(assay_data)
new_names <- mapvalues(x = new_names,
                       from = gene_names$mouse,
                       to = gene_names$human)
rownames(assay_data) <- new_names

seurat_data[["RNA_human"]] <- CreateAssayObject(assay_data,
                                                min.cells = 0,
                                                min.features = 0)

saveRDS(seurat_data, "/tgen_labs/banovich/BCTCSF/PD1_mm_Seurat/PD1_mm_Seurat_merged.Rds")

```

### Visualizing clusters

```{r}

DimPlot(seurat_data,
        group.by = "snn_res.0.5",
        reduction = "umap",
        raster = T,
        cols = cluster_col,
        label = T) &
  coord_fixed(ratio = 1) &
  theme_bw() &
  NoLegend() &
  manuscript_theme

```

### Cell type marker expression

```{r, fig.height=12}

DefaultAssay(seurat_data) <- "RNA"

plot_features <- c()

# Top markers for each cluster
markers <- presto::wilcoxauc(seurat_data,
                             group_by = "snn_res.0.5",
                             assay = "data",
                             seurat_assay = "RNA")

top_markers <- markers %>%  group_by(group) %>% slice_max(order_by = auc, n = 2)

FeaturePlot(seurat_data,
            features = top_markers$feature,
            ncol = 5,
            reduction = "umap",
            raster = T,
            cols = c("gray89", "tomato3")) &
  coord_fixed(ratio = 1) &
  theme_bw() &
  NoLegend() &
  manuscript_theme

```

```{r, fig.width=10, fig.height=12}

top_markers <- markers %>%  group_by(group) %>% slice_max(order_by = auc, n = 5)

# seurat_object, plot_features, group_var, group_colors, column_title, km=5, row.order = NULL
dotplot_heatmap <- create_dotplot_heatmap(seurat_object = seurat_data,
                                          plot_features = unique(top_markers$feature),
                                          group_var = "snn_res.0.5",
                                          group_colors = cluster_col,
                                          column_title = "",
                                          km = 5, row.order = NULL)

```

### Saving top markers to a file

```{r}

top_markers <- markers %>%  group_by(group) %>% slice_max(order_by = auc, n = 20)

write.table(top_markers, "/home/hnatri/PD1_mm/cluster_markers.tsv",
            quote = F, row.names = F, sep = "\t")

```

### Canonical mouse immune markers

```{r, fig.width=10, fig.height=12}

# Mouse immune markers
gs4_deauth()
canonical_markers  <- gs4_get("https://docs.google.com/spreadsheets/d/1ApwXjEVtpPB87al6q3ab8TKvZYJTh3iNH1cuO-A_OoU/edit?usp=sharing")
sheet_names(canonical_markers)
mm_immune_markers <- read_sheet(canonical_markers, sheet = "Mm immune markers")

dotplot_heatmap <- create_dotplot_heatmap(seurat_object = seurat_data,
                                          plot_features = mm_immune_markers$gene_name,
                                          group_var = "snn_res.0.5",
                                          group_colors = cluster_col,
                                          column_title = "",
                                          km = 5, row.order = NULL)

```

### Using sc-ImmuCC to annotate immune cells

```{r, results='hide', warning=F, message=F}

count_data <- LayerData(seurat_data, assay = "RNA_human", layer = "counts")

#scImmuCC_Layered(count = count_data, Non_Immune = FALSE)

# Importing results
scicc_labels <- read.csv("/home/hnatri/PD1_mm/Layer1_scImmuCC_label.csv",
                         row.names = "X")

length(colnames(seurat_data))
length(intersect(scicc_labels$barcodes, colnames(seurat_data)))

seurat_data$scImmuCC_celltype <- mapvalues(x = colnames(seurat_data),
                                           from = scicc_labels$barcodes,
                                           to = scicc_labels$cell_type)

# Plotting
DimPlot(seurat_data,
        group.by = "scImmuCC_celltype",
        reduction = "umap",
        raster = T,
        #cols = scImmuCC_celltype_col,
        label = T) &
  coord_fixed(ratio = 1) &
  theme_bw() &
  NoLegend() &
  manuscript_theme

saveRDS(seurat_data, "/tgen_labs/banovich/BCTCSF/PD1_mm_Seurat/PD1_mm_Seurat_merged.Rds")

```

```{r, fig.height=10, fig.width=8}

DimPlot(seurat_data,
        split.by = "scImmuCC_celltype",
        group.by = "snn_res.0.5",
        ncol = 3,
        reduction = "umap",
        raster = T,
        cols = cluster_col) &
  coord_fixed(ratio = 1) &
  theme_bw() &
  NoLegend()

table(seurat_data$scImmuCC_celltype,
      seurat_data$snn_res.0.5)

```

### Adding annotations from the Google Sheet

```{r}

gs4_deauth()
markers_annotations  <- gs4_get("https://docs.google.com/spreadsheets/d/1iWYBouwQlQboI-rwiujC0QKJ6lq9XeTffbKm2Nz8es0/edit?usp=sharin#g")
sheet_names(markers_annotations)
annotations <- read_sheet(markers_annotations, sheet = "Cluster annotations")

seurat_data$celltype <- mapvalues(seurat_data$snn_res.0.5,
                                  from = annotations$snn_res.0.5,
                                  to = annotations$annotation)

DimPlot(seurat_data,
        group.by = "celltype",
        reduction = "umap",
        raster = T,
        label = T) &
  coord_fixed(ratio = 1) &
  theme_bw() &
  NoLegend()

FeaturePlot(seurat_data,
            features = c("percent.mt_RNA", "nCount_RNA", "nFeature_RNA"),
            reduction = "umap",
            raster = T,
            ncol = 3) &
  coord_fixed(ratio = 1) &
  theme_bw() &
  NoLegend()

saveRDS(seurat_data, "/tgen_labs/banovich/BCTCSF/PD1_mm_Seurat/PD1_mm_Seurat_merged.Rds")

```
